ft: myhighlights popup - show highlights

KarinaMendez2 opened the pull request.

Commit:
initial styles for showing highlights in popup

Commit:
removed console log

Commit:
Merge remote-tracking branch 'origin/master' into show_myhighlights_popup

TomWoodward commented on src/app/content/highlights/components/HighlightsPopUp.tsx:
```diff
@@ -70,9 +83,9 @@ class HighlightsPopUp extends Component<Props> {
           <Styled.MyHighlightsImage src={myHighlightsEmptyImage} />
         </Styled.MyHighlightsWrapper>
-      </Styled.PopupBody>
+      </React.Fragment>
```
you could just return this wrapped in the PopupBody, no need for the fragment and then wrap it in myHighlights

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -299,3 +302,80 @@ export const MyHighlightsImage = styled.img`
   height: ${myHighlightsImageHeight}rem;
   margin-top: ${popupBodyPadding}rem;
 `;
+
```
this file is getting big, can we separate the empty state stuff from the highlights stuff?

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -299,3 +302,80 @@ export const MyHighlightsImage = styled.img`
+    &.highlight.${style.label} {
+      border-left: solid 0.8rem ${style.focused};
+    }
+  `)}
```
this seems like it'd be easier to set the style label as a prop instead of a class and then only render the border for the one style based on that prop instead of putting the css for all of them

TomWoodward commented on src/app/content/highlights/components/ShowMyHighlights.tsx:
```diff
@@ -0,0 +1,39 @@
+            return <Styled.HighlightOuterWrapper>
+              <Styled.HighlightContentWrapper className={'highlight ' + item.style }>
+                <Styled.HighlightContent
+                  key={item.id}
```
key is in the wrong place

TomWoodward commented on src/app/content/highlights/components/ShowMyHighlights.tsx:
```diff
@@ -0,0 +1,39 @@
+                  key={item.id}
+                  dangerouslySetInnerHTML={{__html: item.content}}
+                />
+                { item.note ? <Styled.HighlightNote><p>Note:</p> {item.note}</Styled.HighlightNote> : null }
```
not sure this `p` is semantic, might just want to make it a span

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -299,3 +302,80 @@ export const MyHighlightsImage = styled.img`
+      margin: 0 0.8rem 0 0;
+      overflow: visible;
+    }
+  `)}
```
same here

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -299,3 +302,80 @@ export const MyHighlightsImage = styled.img`
+    .highlight.${style.label} &{
+      background-color: ${style.passive};
+    }
+  `)}
```
and here

Commit:
styles and functionality for
go to top button and loader

TomWoodward committed:
Add loading flag

Commit:
re orginized styles, moved
loader, added timer to highlights by page promise

Commit:
added timers for tests
and loader to work around
highlights, created tests for code coverage in
back to top button

Commit:
Merge remote-tracking branch 'origin/master' into show_myhighlights_popup

Commit:
added test for unmounting
without refs and for loader

TomWoodward commented on src/app/content/highlights/components/ShowMyHighlights.tsx:
```diff
@@ -0,0 +1,88 @@
+
+  };
+
+  public showGoToTop = (bodyElement: HTMLElement) => () => {
```
`updateGoToTop`

Commit:
screenshot updated

TomWoodward commented on src/app/content/highlights/components/ShowMyHighlights.tsx:
```diff
@@ -0,0 +1,88 @@
+  public render() {
+    return (<Styled.ShowMyHighlightsBody ref={this.myHighlightsBodyRef} data-testid='show-myhighlights-body'>
+      <OnScroll callback={this.scrollHandler}/>
+        <Styled.HighlightsChapter>2. Kinematics</Styled.HighlightsChapter>
```
this indentation is kinda messed up

TomWoodward commented on src/app/content/reducer.spec.ts:
```diff
@@ -107,13 +107,17 @@ describe('content reducer', () => {
-      },
-    });
+
+    setTimeout(
```
this is bad, don't add this `setTimeout`

TomWoodward commented on src/app/content/reducer.spec.ts:
```diff
@@ -141,13 +145,16 @@ describe('content reducer', () => {
-      },
-    });
+
+    setTimeout(
```
the `setTimeout` is going to make the test pass unconditionally, nothing inside the callback matters.

KarinaMendez2 commented on src/app/content/highlights/components/HighlightsPopUp.spec.tsx:
```diff
@@ -96,4 +108,205 @@ describe('MyHighlights button and PopUp', () => {
+
+    act(() => { store.dispatch(openMyHighlights()); });
+
+    const target = root.querySelector('[data-testid="show-myhighlights-body"]');
```
the name was wrong... it's written to test the else case inside the "scrollToTop" function when there aren't refs anymore, changed the name.

Commit:
added test for onscroll to
not render without children,
separated highlights pop up
tests from show my highlights test, renamed some functions.

Commit:
lint error fix

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -80,12 +83,15 @@ export const Header = styled(H3)`
   padding: ${popupBodyPadding}rem ${popupPadding}rem;
-  overflow: visible;
+  height: calc(100% - ${headerHeight}rem);
+  overflow-y: scroll;
```
you probably want `auto` not `scroll`, which is the default anyway.

TomWoodward commented on src/app/content/highlights/components/ShowMyHighlights.spec.tsx:
```diff
@@ -0,0 +1,258 @@
+
+    const component = renderer.create(<Provider store={store}>
+      <MessageProvider>
+        <HighlightsPopUp></HighlightsPopUp>
```
you're still rendering the `HighlightsPopUp` component, you should render the `ShowMyHighlights` component directly to simplify these tests

TomWoodward committed:
revert snap

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -63,10 +64,12 @@ export const Modal = styled.div`
+  max-width: ${contentWrapperMaxWidth}rem;
   background: ${theme.color.neutral.base};
-  margin: 3rem;
+  margin: 7.2rem 2.4rem 2.4rem;
```
this isn't centering the modal on large screens

KarinaMendez2 replied:
added a justify-content: center; to the Mask

TomWoodward committed:
Merge remote-tracking branch 'origin/master' into show_myhighlights_popup

TomWoodward committed:
fix spec

TomWoodward committed:
Merge branch 'master' into show_myhighlights_popup

Commit:
removed store and providers from on scroll test
added style to center pop up on large screens
changed some elements in
show my highlights test
to simplify them

Commit:
switched a test location

Commit:
ordered imports
